pkg/frontend/kubeactions/kubeactions.go

```diff
@@ -18,33 +18,51 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/util/retry"
 
 	"github.com/Azure/ARO-RP/pkg/api"
 	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/frontend/kubeactions/gatherer"
 	"github.com/Azure/ARO-RP/pkg/util/restconfig"
 	"github.com/Azure/ARO-RP/pkg/util/version"
 )
 
+// Interface contains all the required methods for kubeactions
 type Interface interface {
-	Get(ctx context.Context, oc *api.OpenShiftCluster, groupKind, namespace, name string) ([]byte, error)
-	List(ctx context.Context, oc *api.OpenShiftCluster, groupKind, namespace string) ([]byte, error)
-	CreateOrUpdate(ctx context.Context, oc *api.OpenShiftCluster, obj *unstructured.Unstructured) error
-	Delete(ctx context.Context, oc *api.OpenShiftCluster, groupKind, namespace, name string) error
-	ClusterUpgrade(ctx context.Context, oc *api.OpenShiftCluster) error
-	MustGather(ctx context.Context, oc *api.OpenShiftCluster, w io.Writer) error
+	Get(ctx context.Context, groupKind, namespace, name string) ([]byte, error)
+	List(ctx context.Context, groupKind, namespace string) ([]byte, error)
+	CreateOrUpdate(ctx context.Context, obj *unstructured.Unstructured) error
+	Delete(ctx context.Context, groupKind, namespace, name string) error
+	ClusterUpgrade(ctx context.Context) error
+	MustGather(ctx context.Context, w io.Writer, g gatherer.Interface) error
 }
 
 type kubeactions struct {
-	log *logrus.Entry
-	env env.Interface
+	log    *logrus.Entry
+	env    env.Interface
+	oc     *api.OpenShiftCluster
+	k8scli kubernetes.Interface
```

Comment:
call this cli to match the rest of our codebase

Reply:
It's "kubernetescli" in the installer to differentiate it: https://github.com/Azure/ARO-RP/blob/ad6b3f781dd2e450307b89cda9ec73e15abce688/pkg/install/install.go#L78

I thought we might move the other kubeactions clients (configcli, dynamiccli) here, so it would need to have a special name.

I do like the "k8s" shorthand but we aren't using it elsewhere.

Reply:
In the first, main commit of this PR, this field is named `cli`. In the second, optional commit, which goes further to move the other clients into `kubeactions`, it's named `kubernetescli`.